### PR TITLE
Capture the time to load the /users/{user_id}/tasks 40 times 

### LINKED ISSUE #310: [TASK] Capture the time to load the /users/{user_id}/tasks 40 times
**Is your feature request related to a problem? Please describe.**
Need to fix the URL to use #-route and be loadable directly (Puppeteer) – capture 40 screenshots and for each one take the time. With cache

**Describe alternatives you've considered**
Tried to fix Hashing without using the react library {HashRouter}

### PR DESCRIPTION
**Pull request by natni**

### CODE REVIEW

**Comment by natni** [comment]
**Description**
Closes #310 

This PR holds a script that captures the time to load the https://nscstrdevusw2thucommon.z5.web.core.windows.net/#/users/{user_id}/tasks URL 40 times with cache. Additionally, the script fixes the URL to use #-route and be loadable directly using Puppeteer

**How to test**
1. Navigate to ui/react-ui from the root folder 
2. Download packages and dependencies needed using `npm install`
3. Start the React app by typing `npm start` _(check if the HashRoute works)_
4. When the app is successfully running on your localhost, please navigate to test/ui from the root folder 
5. Run the _**time-users-id-tasks.js**_ file by typing `time-users-id-tasks.js`
6. You should be able to see how long it took to perform each screenshot(after reloading) including the average time it took for the 40 screenshots.
Image 3-24-21 at 8 18 PM


**Below shows the time it took to load each page**
|**COUNT**|**TIME(sec)**
|---|:---
|1| 1.203
|2| 0.503
|3| 0.541
|4| 0.508
|5| 0.523
|6| 0.537
|7| 0.517
|8| 0.525
|9| 0.521
|10| 0.531
|11| 0.569
|12| 0.516
|13| 0.512
|14| 0.536
|15| 0.517
|16| 0.515
|17| 0.538
|18| 0.544
|19| 0.535
|20| 0.55
|21| 0.497
|22| 0.5
|23| 0.52
|24| 0.516
|25| 0.523
|26| 0.534
|27| 0.513
|28| 0.553
|29| 0.617
|30| 0.504
|31| 0.527
|32| 0.518
|33| 0.539
|34| 0.534
|35| 0.526
|36| 0.545
|37| 0.507
|38| 0.513
|39| 0.603
|40| 0.516

**The average time to load is **0.55****







TIME SHEET

|**DATE**|**ACTIVITY**|**TIME**|
|---|:---|:---|
|03/21| Research on HashRouting and implementation |2 hours|
|03/22| Writing a script to calculate the time to load the URL and capture time |2 hours |
|03/24| Final touch-ups and writing descriptions  |2 hours |

File: test/ui/time-users-id-tasks.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+const puppeteer = require("puppeteer");
+
+(async () => {
+    const uiLocation = "https://nscstrdevusw2thucommon.z5.web.core.windows.net/#/users/6/tasks ";
+    const browser = await puppeteer.launch();   
+    var startDate = new Date();
+    var totalSeconds = 0;     
+    var startTime = startDate.getTime();
+
+    for(var i=0; i<40; i++){
```
Comment (chonger878) [inline test/ui/time-users-id-tasks.js:10]: I am a little concerned about the hardcoded value here.

**Review by chonger878 — COMMENTED** [review]
The code is easy to follow for me and I understand most of the logic.  It's just that little hard coded value that I am concerned about since my understanding is that we're supposed to avoid them...? I might be wrong.

File: test/ui/time-users-id-tasks.js (hunk 1 above)
Comment (chonger878) [inline test/ui/time-users-id-tasks.js]: For mine, I did a global variable like NUM_TIMES = 40, since it won't necessarily be always 40 times.

Comment (natni) [reply test/ui/time-users-id-tasks.js]: Fixed it thanks!

File: test/ui/time-users-id-tasks.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    var startTime = startDate.getTime();
+
+    for(var i=0; i<captureTime; i++){
+        const page = await browser.newPage();
+        console.log(`Page opened, going to ${uiLocation}`);
+
+        await page.goto(uiLocation)
+
+        console.log(`Navigated to ${uiLocation}, taking screenshot`);
+            await page.screenshot();
```
Comment (unicornsleuth) [inline test/ui/time-users-id-tasks.js:16]: Save the screenshots to the repo so we can be sure this is working! I definitely had a lot of screenshots of "Loading..." or the spinning circle, so I had to change my `page.got(url)` to `page.goto(url, waitUntil="networkidle2")` to fix it.

**Review by unicornsleuth — COMMENTED** [review]
Add .DS_Store to the .gitignore
Save the screenshots to the repo (don't need to upload to Azure) so we can verify that it's waiting until the data loads!

**Review by chonger878 — APPROVED** [review]
I guess the screenshots are needed to confirm that the time was captured? I assume it's strongly recommended?

File: test/ui/time-users-id-tasks.js (hunk 2 above)
Comment (natni) [reply test/ui/time-users-id-tasks.js:16]: Toddy told me we don't need to store the screenshots that's why

**Review by unicornsleuth — APPROVED** [review]

**Pull request merged**